src/resolve/resolved-dns-server.c

```diff
@@ -609,7 +609,10 @@ static void dns_server_hash_func(const DnsServer *s, struct siphash *state) {
 
         siphash24_compress(&s->family, sizeof(s->family), state);
         siphash24_compress(&s->address, FAMILY_ADDRESS_SIZE(s->family), state);
+        siphash24_compress(&s->port, sizeof(s->port), state);
         siphash24_compress(&s->ifindex, sizeof(s->ifindex), state);
+        if (s->server_name)
+                siphash24_compress(s->server_name, strlen(s->server_name), state);
```

Comment:
`siphash24_string` ?

Reply:
I do not know well about `siphash24_string()` but it seems not to be intended to use here... At many places, we use `siphash24_compress()` with `strlen()`.

Reply:
siphash24_string() does almost the same thing as the open-coded version here. The difference is that the siphash24_string() included the terminating NUL in the calculation. This means that the result is different for NULL and "". This difference does not matter here, or maybe it is even better to make those cases different. Either way, using `siphash24_string()` is appropriate.

Reply:
Hmm?? `siphash24_string()` is a wrapper of `siphash24()`, not `siphash24_compress()`.

Maybe, we should introduce another thin wrapper `siphash24_compress_string()` later.

Reply:
See https://github.com/yuwata/systemd/commit/f10b4bee894b05c1ac105679aa37051adbb72a59 and https://github.com/yuwata/systemd/commit/ecd54dc3ba927fa1773dddd1658844f1f0a4af29.

Reply:
Right, I was confused. Never mind then.